src/interp.rs

```diff
@@ -290,16 +289,15 @@ impl<'a, 'tcx> Machine<'a, 'tcx> {
     /// flow. A `Terminator` determines the next `BasicBlock` that should be jumped to
     /// during execution.
     fn eval_terminator(&mut self,
-                       terminator: &Terminator<'tcx>) {
+                       terminator: &Terminator<'tcx>)
+                       -> Option<BasicBlock> {
```

Comment:
I can't immediately see how this can return `None` -- any clues?

Reply:
If a function is of type `!` it won't have a return block and will instead return None. This is how rust deals with non-termination or exiting of programs, so I need to account for it here.

Reply:
That's true, but I don't see how that's relevant to `-> Option<BasicBlock>` :) What I meant was: from this diff I can't see anywhere where `None` is returned. Pointers welcome :)

Reply:
It's mainly for the `TerminatorKind::Return` arm on line 296. This returns the RA stored against the `Frame`, this could be `None` if the function call is non-terminating

Reply:
Aha, I see!